open_spiel/python/mfg/algorithms/softmax_policy_test.py

```diff
@@ -0,0 +1,103 @@
+# Copyright 2019 DeepMind Technologies Ltd. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests for softmax_policy."""
+
+from absl.testing import absltest
+from absl.testing import parameterized
+
+from open_spiel.python import policy
+from open_spiel.python.mfg import value
+from open_spiel.python.mfg.algorithms import best_response_value
+from open_spiel.python.mfg.algorithms import distribution
+from open_spiel.python.mfg.algorithms import greedy_policy
+from open_spiel.python.mfg.algorithms import softmax_policy
+from open_spiel.python.mfg.algorithms import policy_value
+from open_spiel.python.mfg.games import crowd_modelling  # pylint: disable=unused-import
+import pyspiel
+
+
+class SoftmaxPolicyTest(parameterized.TestCase):
+
+  @parameterized.named_parameters(('python', 'python_mfg_crowd_modelling'),
+                                  ('cpp', 'mfg_crowd_modelling'))
+  def test_softmax(self, name):
+    """Check if the softmax policy works as expected.
+
+    The test checks that:
+    - uniform prior policy gives the same results than no prior.
+    - very high temperature gives almost a uniform policy.
+    - very low temperature gives almost a deterministic policy for the best action.
+
+    Args:
+      name: Name of the game.
+    """
+
+    # Checks that a greedy policy with respect to an optimal value is an optimal policy.
+    game = pyspiel.load_game(name)
+    uniform_policy = policy.UniformRandomPolicy(game)
+    dist = distribution.DistributionPolicy(game, uniform_policy)
+    br_value = best_response_value.BestResponse(
+        game, dist, value.TabularValueFunction(game))
+    br_val = br_value(game.new_initial_state())
```

Comment:
It would be great if we can name *_val to *_init_val. When there is also *_value, it will be more explicit and easier to read.

Reply:
Done

Reply:
Done